permission_prompt/explariner_navigation.md

```diff
@@ -0,0 +1,89 @@
+
+- **Author**: lyf@google.com
+- **Created**: 2023-11-24
+- **Last Updated**: 2024-01-26
+
+## A problem
+
+Currently, any public page can make a request as if they were the user's machine. This allows a malicious attacker to use a public web page to attack network endpoints on a user's private network. These endpoints (router, printer, IoT devices, ...) are often badly protected because they do not expect to receive requests from the web at large.
+
+To mitigate this threat, we have been working on the Private Network Access (PNA) spec, which prevents public pages from making requests to private network endpoints, unless the endpoint opts into receiving the request from the public origin. This model works if the origin can be validated, i.e. if the public page is served over HTTPS. To support PNA, we want to remove the ability for public HTTP pages to make requests on private network endpoints.
+
+This means that public pages served on HTTP that embed resources from the private network need to upgrade to HTTPS. However, there is currently no good solution for encryption on the private network, meaning that many of those subresources are served over HTTP and cannot upgrade to HTTPS. They cannot be embedded in an HTTPS page due to mixed contents checks.
+
+To solve this issue, we have introduced a permission prompt to relax mixed contents checks when fetching resources from the private network. However, it only covers subresources but not iframes. We want to extend this mechanism to navigation.
+
+## Deprecated solutions
+### Service worker
+
+In the [previous discussion](https://github.com/WICG/private-network-access/issues/83), we planned to introduce the permission for
+workers, so that it can be used for non-fetch requests. However, the mixed 
+content check actually happens before service workers, which means it won't work.
+
+## A solution
+
+In this case, we would like to introduce a new [Content Security Policy](https://www.w3.org/TR/CSP3/) to let 
+all documents declare their IP address space.
+
+### New [directive](https://www.w3.org/TR/CSP3/#framework-directives)
+
+Two new Content Security Policy directives are introduced for Private Network Access, 
+`private-address-space` and `local-address-space`.
+
+> Notes: `localhost`, `::1/128` and `127.0.0.1` hosts are regard as secure in mixed 
+> content check, so that we don't need to relax. Benchmarking address 198.18.0.0/15 is regarded as a local IP address in Private Network Access specification but not potentially trustworthy origin in secure contexts spec.
+
+### Policy delivery
+
+[The same as all the other Content Security Policies](https://www.w3.org/TR/CSP3/#policy-delivery), private/local IP address 
+space can be declared as either a HTTP header or a HTML `<meta>` element.
+
+#### HTTP header
+
+To declare private IP addresses:
+
+```text
+Content-Security-Policy: private-address-space private_hosts
```

Comment:
We can perhaps follow the MDN syntax. https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/frame-src

```suggestion
To declare one or more private IP addresses:

```text
Content-Security-Policy: private-address-space <private_host>
Content-Security-Policy: private-address-space <private_host> <private_host>
```